activemodel/lib/active_model/callbacks.rb

```diff
@@ -127,26 +129,34 @@ def define_model_callbacks(*callbacks)
     private
 
       def _define_before_model_callback(klass, callback)
-        klass.define_singleton_method("before_#{callback}") do |*args, &block|
-          set_callback(:"#{callback}", :before, *args, &block)
+        klass.define_singleton_method("before_#{callback}") do |*args, **options, &block|
+          verify_options(__method__, options)
+          set_callback(:"#{callback}", :before, *args, **options, &block)
         end
       end
 
       def _define_around_model_callback(klass, callback)
-        klass.define_singleton_method("around_#{callback}") do |*args, &block|
-          set_callback(:"#{callback}", :around, *args, &block)
+        klass.define_singleton_method("around_#{callback}") do |*args, **options, &block|
+          verify_options(__method__, options)
+          set_callback(:"#{callback}", :around, *args, **options, &block)
         end
       end
 
       def _define_after_model_callback(klass, callback)
-        klass.define_singleton_method("after_#{callback}") do |*args, &block|
-          options = args.extract_options!
+        klass.define_singleton_method("after_#{callback}") do |*args, **options, &block|
+          verify_options(__method__, options)
           options[:prepend] = true
           conditional = ActiveSupport::Callbacks::Conditionals::Value.new { |v|
             v != false
           }
           options[:if] = Array(options[:if]) << conditional
-          set_callback(:"#{callback}", :after, *(args << options), &block)
+          set_callback(:"#{callback}", :after, *args, **options, &block)
+        end
+      end
+
+      def verify_options(method, options)
```

Comment:
This name is waaaay too generic for us to add everywhere that `AMo::Callbacks` gets included.

It doesn't actually use any instance state, though, so instead of giving it a big ugly name, let's just move it onto the module itself.

Reply:
Do you mean as a class method (on the module) so it'd be called like this?
```ruby
Callback.verify_options(__method__, options)

Reply:
Yep :+1: